app/javascript/components/map/MapPopup.vue

```diff
@@ -110,6 +100,16 @@ export default {
       this.randomKey += 1;
       this.data.datasets = [];
 
+      const ids =[]
+      this.chartData.forEach(item => ids.push(item.id))
+
+      for (const [key, value] of Object.entries(CHART_COLOURS)) {
+        // if using ids key has to be changed to Number type --> Number(key) https://stackoverflow.com/a/49550265/10897687
+        if (ids.includes(Number(key))) {
+          this.colors.push(value)
+        }
+      }
```

Comment:
Could this be simplified to something like this?

```
this.chartData.forEach(item => {
    const colour = CHART_COLOURS[item.id]
    this.colors.push(colour)
}
```

Reply:
Re: https://github.com/unepwcmc/action-agenda/pull/106#discussion_r891048100 the second line could be `const colour = CHART_COLOURS[item.id] || "#e3e3e3"` (or whatever default colour)